### PR TITLE
Fix bugs in mainmenu

### PR DESCRIPTION
**Pull request by kilbith**
Last attempt to fix this annoying issue : https://github.com/minetest/minetest/issues/2548

With suggestions of @rubenwardy and @ShadowNinja applied.

### CODE REVIEW

**Comment by rubenwardy** [comment]
:+1: all the issues I mentioned have been fixed.

File: builtin/mainmenu/tab_server.lua [hunk 1]
```diff
@@ -67,31 +67,20 @@ local function main_button_handler(this, fields, name, tabdata)
...
+			core.setting_set("creative_mode", creative_mode)
+		else
+			menu_worldmt(selected, "creative_mode", core.setting_get("creative_mode"))
+		end
+
+		local enable_damage = menu_worldmt(selected, "enable_damage")
+		if enable_damage then
+			core.setting_set("enable_damage", enable_damage)
+		else
+			menu_worldmt(selected, "enable_damage", core.setting_get("enable_damage"))
```
Comment (nerzhul) [inline builtin/mainmenu/tab_server.lua]: Merge this code with...

File: builtin/mainmenu/tab_singleplayer.lua [hunk 2]
```diff
@@ -105,31 +105,20 @@ local function main_button_handler(this, fields, name, tabdata)
...
+			core.setting_set("creative_mode", creative_mode)
+		else
+			menu_worldmt(selected, "creative_mode", core.setting_get("creative_mode"))
+		end
+
+		local enable_damage = menu_worldmt(selected, "enable_damage")
+		if enable_damage then
+			core.setting_set("enable_damage", enable_damage)
+		else
+			menu_worldmt(selected, "enable_damage", core.setting_get("enable_damage"))
```
Comment (nerzhul) [inline builtin/mainmenu/tab_singleplayer.lua]: ... this code, using a function

**Comment by nerzhul** [comment]
Except the code which could be factorized, it's okay for me

**Comment by kilbith** [comment]
@nerzhul Done. Better ?

**Comment by nerzhul** [comment]
@kilbith yes ! thanks

**Comment by est31** [comment]
:+1: you merge if you +1ed nrz.

**Comment by est31** [comment]
85f3d575ec3d99ef2ce680d4a2546e4d31327d83

**Pull request closed without merging**